src/gretel_trainer/relational/multi_table.py

```diff
@@ -767,22 +802,51 @@ def _train_synthetics_models(self, training_data: dict[str, Path]) -> None:
                     self._extended_sdk,
                 )
 
-        # TODO: consider moving this to before running the task
-        archive_path = self._working_dir / "synthetics_training.tar.gz"
-        for table_name, csv_path in training_data.items():
-            add_to_tar(archive_path, csv_path, csv_path.name)
-        self._artifact_collection.upload_synthetics_training_archive(
-            self._project, str(archive_path)
-        )
-
     def train(self) -> None:
-        """Train synthetic data models on each table in the relational dataset"""
+        """
+        DEPRECATED: Please use `train_synthetics` instead.
+        """
+        logger.warning(
+            "This method is deprecated and will be removed in a future release. "
+            "Please use `train_synthetics` instead."
+        )
         tables = self.relational_data.list_all_tables()
         self._synthetics_train = SyntheticsTrain()
 
         training_data = self._prepare_training_data(tables)
         self._train_synthetics_models(training_data)
 
+    def train_synthetics(
+        self,
+        *,
+        only: Optional[list[str]] = None,
+        ignore: Optional[list[str]] = None,
+    ) -> None:
+        """
+        Train synthetic data models for the tables in the tableset,
+        optionally scoped by either `only` or `ignore`.
+        """
+        only, ignore = self._get_only_and_ignore(only, ignore)
+
+        all_tables = self.relational_data.list_all_tables()
+        include_tables = []
+        omit_tables = []
+        for table in all_tables:
+            if skip_table(table, only, ignore):
+                omit_tables.append(table)
+            else:
+                include_tables.append(table)
```

Comment:
Maybe I'm not at the part of this PR where it becomes clear... I see, opaquely here, that we use skip_table to add to the list of omit_tables.  But then it looks like we otherwise throw the table onto the include list.  Hoping it becomes clear somewhere else how we only have the only's here.  If it does, this is just a bookmark comment for me and can be safely ignored :wave:

Does the only filtering happen in list_all_tables?

Reply:
The [skip_table function](https://github.com/gretelai/trainer/blob/main/src/gretel_trainer/relational/core.py#L667-L676) got added in a separate branch and is being reused here. uses the `only` and `ignore` lists to determine if the provided `table` should be skipped or not.

(Should have a docstring.)

Reply:
Thanks for clarifying.